apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/Numeric.cs

```diff
@@ -0,0 +1,482 @@
+﻿// Copyright 2020 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Numerics;
+using System.Text.RegularExpressions;
+
+namespace Google.Cloud.Spanner.V1
+{
+    // TODO: Implement IFormattable?
+
+    /// <summary>
+    /// Representation of the Spanner NUMERIC type, which has 38 digits of precision,
+    /// and a fixed scale of 9 decimal places to the right of the decimal point
+    /// </summary>
+    public struct Numeric : IEquatable<Numeric>, IComparable<Numeric>, IComparable
```

Comment:
I'm nervous of *just* calling this Numeric. I think I'd prefer to call it `SpannerNumeric`. This fits in with the pattern of [SqlDecimal](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqltypes.sqldecimal?view=dotnet-plat-ext-3.1) etc.

@amanda-tarafa do you have thoughts either way?

Reply:
I should've explained my reasoning for this. I thought that if we were to ever move this class to a common place and share it between Spanner and BigQuery then it might be easier if we call it Numeric now. But I don't know if that'll ever happen.

Reply:
If it ever happened, it would be a breaking change anyway - and arguably having a Spanner-specific name means that a general "Numeric" type can be *added* without introducing naming collisions, allowing an easier migration path.

Reply:
Ok makes sense. I'm happy for this to be called SpannerNumeric. Probably won't get around to doing this till next week though.

In the meantime, if @amanda-tarafa has any thoughts, please chime in.

Reply:
I agree that for the meantime it makes more sense to call it SpannerNumeric, it goes well with SpannerStruct and with the more general SpannerDbType.
I haven't looked at this PR but as far as I can see this is just adding the numeric class, right? The conversion hooks are yet to be added? I might be misunderstanding something, so take this with a pinch of salt.

Reply:
I found some time and renamed it to SpannerNumeric.

@amanda-tarafa, I'm just adding the class in this PR. I'll send a separate PR that adds the conversion hooks.